web-gateway/src/main/java/controllers/AbstractController.java

```diff
@@ -47,17 +50,21 @@ public AbstractController(MessagesApi messagesApi, UserService userService) {
     }
 
     protected CompletionStage<Nav> loadNav(Optional<UUID> userId) {
-        return userService.getUsers(Optional.of(0), Optional.of(10)).invoke().thenApply(users -> {
-            Optional<User> currentUser = userId.flatMap(id -> {
-                for (User u: users.getItems()) {
-                    if (u.getId().equals(id)) {
-                        return Optional.of(u);
-                    }
+        CompletionStage<User> u = userService.getUser(userId.get())
+                .handleRequestHeader(authenticate(userId.get())).invoke();
+        CompletionStage<PaginatedSequence<User>> users = userService.getUsers(Optional.of(0), Optional.of(10))
+                .handleRequestHeader(authenticate(userId.get())).invoke();
```

Comment:
When no user has logged in yet, the list of `users` will not be accessible. This authentication is not required.